### PR TITLE
Aligning version constraint with module 5.6.x

### PR DESCRIPTION
**Pull request by arnaudlh**
Aligning version number with module 5.6.x for random provider

### CODE REVIEW

File: caf_solution/main.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
     // azuread version driven by the caf module
     random = {
       source  = "hashicorp/random"
-      version = "~> 3.3.2"
+      version = "~> 3.3.0"
```
Comment (segraef) [inline caf_solution/main.tf:7]: Tested with `3.3.1` works fine.

File: caf_launchpad/main.tf [hunk 2]
```diff
@@ -4,7 +4,7 @@ terraform {
     // azuread version driven by the caf module
     random = {
       source  = "hashicorp/random"
-      version = "~> 3.1.0"
```
Comment (segraef) [inline caf_launchpad/main.tf:7]: Tested with `3.3.1` works fine.

File: caf_solution/main.tf (hunk 1 above)
Comment (arnaudlh) [reply caf_solution/main.tf:7]: Constraints allow last digit update so will tolerate bugfixes, currently in CI/CD with 3.3.2 as per https://github.com/aztfmod/terraform-azurerm-caf/pull/1185/files

File: caf_launchpad/main.tf (hunk 2 above)
Comment (arnaudlh) [reply caf_launchpad/main.tf:7]: Constraints allow last digit update so will tolerate bugfixes, currently in CI/CD with 3.3.2 as per https://github.com/aztfmod/terraform-azurerm-caf/pull/1185/files

**Review by segraef — APPROVED** [review]
image
lgtm, remember aztfmod is at `3.3.1` but fine so far

**Pull request merged**